### DISCUSSION TITLE
Transform / make webp/avif versions of ie. public/media folder upon build?

### DISCUSSION THREAD

**Discussion by hjelmdal** [JonasKruckenberg/imagetools#517] [Q&A]
Hi, 
first of all, please bear with me, if this is a silly question, but I am very new to this :-) 

So, I've tried out the library on my vue / vite app - and I really love it, although I have run into some problems. 

1. I can not import directly from the public folder in my components
2. I need to do TS ignore on imports because TSC says that files are not resolved, when querystring is added. 
Apart from that, it works really well!

But.. One thing I would like to do, is to integrate this library to just take my entire /public/media folder, and make wepb or avif versions of all images there, if they can optimize the original filesize (in a way like vite-image-optimizer does)
![image](https://user-images.githubusercontent.com/5920880/225567145-1073e4b3-6d49-4ee2-8738-e9bd63024e4f.png)

I don't know whether this is an option or not, but to me it would be a really nice feature, so that I can just serve the right version based on the browser support of webp/avif and replace the .png / jpg with avif/webp if supported. 

I hope you get the point, and hope that you can tell me if this is possible or not - sorry if I missed out on something :-)

**Comment by benmccann** [comment]
For Svelte projects, I use https://github.com/bluwy/svelte-preprocess-import-assets
